### PR TITLE
KAFKA-9423: Refine layout of configuration options on website and make individual settings directly linkable

### PR DESCRIPTION
**Pull request by soenkeliebau**
Test strategy:
Built the site and served from an Apache httpd docker container. Used w3c validator to ensure generated HTML is valid.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mimaison** [comment]
Thanks @soenkeliebau! I'll try to take a look this week.

**Comment by mimaison** [comment]
For reference, this is how this renders:
image

I feel like the table with the details is a bit oversized. We could also keep the outer list to keep the alternating background colors for the settings. What do you think?

**Comment by soenkeliebau** [comment]
I've played around a little (and cursed about HTML and CSS quite a bit in the process) and have ended up with this suggestion:
image

I've kept it a table, even if invisible because I personally feel that having the values in the second column aligned makes it more readable.

We'd lose this if we made it a list because the length of the words in the first column is different:
```
Type: itaeiae
Update Mode: iaeiae
```

The code that generates this is butt ugly at the moment because it needs a lot of styles manually set to the HTML output. I wanted to wait for feedback on this "design" before I figured out how to improve that.

**Comment by mimaison** [comment]
That looks good to me. Let's see what other people think

**Comment by soenkeliebau** [comment]
One question that occurred to me, currently the name of the setting is formatted as `<h4>` because I thought that would add a linkable anchor automatically. That wasn't the case though and I am now manually adding the anchor, so we might downgrade it from h4 to just bold or something similar.
Not sure if there are any benefits or drawbacks to either approach tbh. I'm not an HTML guy..

**Comment by kguelseven** [comment]
Looks good to me too. Suggestions for minor improvements:
- make the alternating items (divs) more distinct by using a darker gray or a separator line in between
- more padding inside divs to make content come out more
- consider using "back-to-back" alignment (align: right -> <- align: left) within the tables. This makes the listing more readable imo.

**Comment by hachikuji** [comment]
This looks good to me too. +1 on @kguelseven's suggestion for alternating colors or a separator between lines.

**Comment by soenkeliebau** [comment]
How about this?

image

**Comment by kguelseven** [comment]
Nice. It's more readable now.

**Comment by soenkeliebau** [comment]
I've pushed an updated version that produces the image (or a close resemblance of it) that we last agreed upon.
I probably violate 9 out of the 10 golden rules of webdesign with the output, but I don't know any better and frankly have spent more time on html and css pages today than I care to.
Comments more than welcome :)

Also: is everybody fine with that little line in there that replaces `\n` with `<br>` ? Didn't see any comments on that yet, but I don't want to just sneak it in!

**Comment by hachikuji** [comment]
@soenkeliebau Thanks, I really appreciate your efforts here. I will leave @mimaison to approve this, but the latest change looks good to me.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: We should move this CSS to the website CSS file instead of having it inline here. The CSS file is in kafka-site, see https://github.com/apache/kafka-site/blob/asf-site/css/styles.css#L266.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: Can we add line return after HTML block like tables, rows, list items so the generated HTML stays readable?

**Review by mimaison — CHANGES_REQUESTED** [review]
Thanks for progressing this issue! Let's use the CSS file instead of having the style inline.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (soenkeliebau) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: Ok, I'll add it to that file - was hoping to sneak past without having two separate pull requests ;)

I'll just add a class "config-list" and put everything in there?

**Comment by joel-hamill** [comment]
this looks great to me. i actually prefer the first iteration without alternating grey (https://github.com/apache/kafka/pull/7955#issuecomment-575338405), but honestly either iteration seems like an improvement. my argument for leaving out the grey would be that i assume most people are probably navigating this content by using `ctrl + f` rather than manually scrolling through the entire table.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (soenkeliebau) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: I've added some newlines to make it more readable. It's still not great due to missing indentation, but I felt that would be taking it too far.

**Comment by mimaison** [comment]
Thanks @soenkeliebau for the quick update. I'm traveling for the rest of the week, I'll take a look early next week

**Comment by soenkeliebau** [comment]
Hi @mimaison, no rush, whenever you find the time!

**Comment by mimaison** [comment]
The layout looks good to me but before merging I've asked the IBM Design team to take a quick look. They'll also be able to advise on the CSS.
I'm giving them a few days to check it out and hopefully we can get this in for 2.5.0.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: This makes stopbugs complain ... We should use `%n` in the format string instead of `\n`

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: No need for the extra `:`, `addConfigDetail()` already inserts it

**Review by mimaison — CHANGES_REQUESTED** [review]
Thanks @soenkeliebau. It looks good, I just spotted 2 small issues. Once fixed I'm happy to merge

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (soenkeliebau) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: Done

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java
Comment (soenkeliebau) [reply clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java]: Done

**Comment by soenkeliebau** [comment]
Addressed those two issues.

**Review by mimaison — APPROVED** [review]
Thanks for the update, LGTM

**Comment by mimaison** [comment]
The test failures are unrelated, merging

**Comment by mimaison** [comment]
@mumrah can I merge this in 2.5?

**Comment by mimaison** [comment]
Merged in 2.5, thanks again @soenkeliebau for making this change

**Pull request merged**
